modules/developers-guide/pages/graphql-oriented-using.adoc

```diff
@@ -0,0 +1,1062 @@
+= Using the Stargate GraphQL API (GraphQL-oriented)
+
+Stargate is a data gateway deployed between client applications and a database.
+The GraphQL API plugin that exposes CRUD access to data stored in Cassandra tables.
+
+include::partial$graphql-oriented-desc.adoc[]
+
+include::partial$GQLAPIBlogPost.adoc[]
+
+// tag::prereqsList[]
+include::partial$prereqs.adoc[]
+// end::prereqsList[]
+
+// not in astra yet
+//include::developers-guide:partial$astra_tip.adoc[]
+
+// tag::getDockerImage[]
+include::partial$docker_pull.adoc[tag=3x]
+// end::getDockerImage[]
+
+// tag::startDocker[]
+include::partial$docker_run.adoc[tag=3x]
+// end::startDocker[]
+
+== Using the Auth API to generate an auth token
+In order to use the Stargate Document API, an authorization token must be
+generated to access the interface. A REST API token is used for this purpose.
+
+// generate the auth token
+include::partial$gen_auth_token.adoc[]
+
+You will need to add this token to the GraphQL Playground in order to authorize
+your GraphQL requests. Copy the value after `"authToken"` to use later.
+
+include::graphql.adoc[tag=UsingGraphQLPlayground]
+
+include::graphql.adoc[tag=UsingPostman]
+
+== Creating or dropping schema
+
+In order to use the GraphQL API, you must xref:graphql-oriented-using.adoc##_deploying_schema[deploy schema]
+that defines the types, mutations, and queries.
+However, a keyspace, a container that stores the data, must first be created.
+
+// tag::CreateKS[]
+=== Creating a keyspace
+
+Before you can start using the GraphQL API, you must first create a Cassandra
+keyspace in your database.
+
+// LLP: need to check how to write this - it's migration...If you are connecting to a Cassandra database with existing schema, you can skip this step.
+
+Inside the GraphQL playground, navigate to
+http://localhost:8080/graphql-schema[http://localhost:8080/graphql-schema]
+and create a keyspace by executing the following mutation:
+
+[source, plaintext]
+----
+include::example$graphql/createKeyspace.graphql[]
+----
+For each keyspace created in your Cassandra schema, a new path is created under
+the `graphql-path` root (default is: `/graphql`). For example, the mutation just
+executed creates a path `/graphql/library` for the `library` keyspace when
+Cassandra creates the keyspace.
+
+Add the auth token to the HTTP Headers box in the lower lefthand corner:
+
+[source, plaintext]
+----
+{
+  "X-Cassandra-Token":"bff43799-4682-4375-99e8-23c8a9d0f304"
+}
+----
+
+[IMPORTANT]
+====
+Notice that the key for this JSON token is different than the value that the
+generate token has. It is `X-Cassandra-Token`, not `auth-token`.
+====
+
+Now run the mutation to create the keyspace. You should see a return value of:
+
+[source, plaintext]
+----
+include::example$result/gql_createKeyspace.result[]
+----
+// end::CreateKS[]
+
+// tag::CreateUDT[]
+=== Creating a user-defined type (UDT)
```

Comment:
There should be a high-level explanation of schema deployment before we dive into the details. Seeing the `type Address` snippet below, I think my first reaction as a developer would be "where does that go?". The answer only comes in the "Deploying schema" section way further down.

Reply:
I get what you are saying. But to discuss deploying, you have to show an example. I would be frustrated to have all that schema shown to me without understanding what it is. I guess I could put in a TIP that says "want to understand deployment first? go here" and send them down the page.

Reply:
I like the tip idea

Reply:
Added a tip and some explanation.